panels/updates/cc-tariff-editor.c

```diff
@@ -257,8 +256,18 @@ update_tariff (CcTariffEditor *self,
   /* 2. Allowed downloads period */
   allowed_start_utc = g_date_time_new_from_unix_utc (self->seconds_to_start);
   allowed_end_utc = g_date_time_new_from_unix_utc (self->seconds_to_end);
```

Comment:
Seems to me that the problem is that `seconds_to_start` and `seconds_to_end` are in the local timezone (because they’re directly calculated from the hours/minutes the user inputs in the UI), but they’re being converted to a `GDateTime` using `g_date_time_new_from_unix_utc()`. Would it work to do
```
allowed_start = g_date_time_new_from_unix_local (self->seconds_to_start);
allowed_end = g_date_time_new_from_unix_local (self->seconds_to_end);
```

I was wondering if that opens this code up to potential failure (`g_date_time_new_from_unix_local()` returning `NULL`) if, for example, the user has set downloads to start at midnight, but they’re in a positive timezone (UTC+x) — but I don’t think it can fail, since `GDateTime` supports UTC instants down to the year 1AD, whereas a positive timezone on a UNIX timestamp of 0 would result in a time around December 1969.

It would still be useful to add `g_assert (allowed_start != NULL)` (etc.) just to make it clearer that the error handling here has been thought about.